cmd/manager/main.go

```diff
@@ -127,6 +132,18 @@ func main() {
 		os.Exit(1)
 	}
 
+	// Setup Scheme for all monitoring resources
+	if err := monitoringv1.AddToScheme(mgr.GetScheme()); err != nil {
+		log.Error(err, "")
+		os.Exit(1)
+	}
+
+	// Setup Scheme for all grafana resources
+	if err := grafanav1alpha1.AddToScheme(mgr.GetScheme()); err != nil {
```

Comment:
Does adding the scheme mean that in order to be able to start the operator now the grafana-related CRDs have to be there beforehand? I think in case of prometheus the CRDs are already there because OpenShift clusters install them but I think it's not the case for Grafana-related CRDs.
I understand we require the CRDs even if we are not using the functionality.

If so, I guess we should provide documentation team with steps to document the deployment of grafana-related CRDs?

Reply:
No. Adding the scheme does no require the CRD to be registered in the cluster.

Reconcilers ask the cluster if the CRD's are installed. If not, CRs are safely discarded with a log message:

```
{"level":"info","ts":1591386801.247215,"logger":"controller_apimanager","msg":"Install grafana-operator in your cluster to create grafanadashboards objects","APIManager Controller":"apimanager1","Error creating grafanadashboard object":"backend"}
```

For that kind of checks, a new discovery client is being added to the base reconciler.

Here you can see the reconcilers checking for specific CRD to be available in the cluster: 
https://github.com/3scale/3scale-operator/pull/333/files?file-filters%5B%5D=.go&file-filters%5B%5D=.tpl#diff-bd922f847ce6c805b996495ed4e5a612R126